cornice/validators/_colander.py

```diff
@@ -62,28 +62,19 @@ def validator(request, schema=None, deserializer=None, **kwargs):
     if deserializer is None:
         deserializer = extract_cstruct
 
-    if schema is None:
-        raise TypeError('This validator cannot work without a schema')
-
-    schema = _ensure_instantiated(schema)
-    cstruct = deserializer(request)
-    try:
-        deserialized = schema.deserialize(cstruct)
-        request.validated.update(deserialized)
-    except colander.Invalid as e:
-        translate = request.localizer.translate
-        error_dict = e.asdict(translate=translate)
-        for name, msg in error_dict.items():
-            prefixed = name.split('.', 1)
-
-            if len(prefixed) == 1:
-                location = name
-                field = ''
-            else:
-                location = prefixed[0]
-                field = prefixed[1]
-
-            request.errors.add(location, field, error_dict[name])
+    if schema is not None:
```

Comment:
I come from the days of *Structured Programming* and tend to adhere to the “Single entry, single exit” (SESE) paradigm — it's just the way my brain works.  In this case, there's so little code that it doesn't really matter either way.  I'll change it...